lib/Db/MessageMapper.php

```diff
@@ -487,9 +487,17 @@ public function deleteAll(Mailbox $mailbox): void {
 			->from($this->getTableName())
 			->where($messageIdQuery->expr()->eq('mailbox_id', $deleteRecipientsQuery->createNamedParameter($mailbox->getId())));
 
+		$cursor = $messageIdQuery->execute();
+		$messageIds = $cursor->fetchAll();
```

Comment:
```suggestion
		$messageIds = $cursor->fetchFirstColumn();
```

maybe? no array_map then.

Reply:
that didn't exist with Nextcloud 20 but only after the migration of dbal in 21